streams/src/main/java/org/apache/kafka/streams/processor/internals/InternalTopologyBuilder.java

```diff
@@ -113,6 +113,9 @@
 
     private final QuickUnion<String> nodeGrouper = new QuickUnion<>();
 
+    // state stores whose source topics are reused as their changelog
+    private final Set<String> optimizedSourceTables = new HashSet<>();
```

Comment:
We need to keep track of which tasks have optimized source tables, since they are marked as `loggingDisabled`